writable-stream-abort-signal-explainer.md

```diff
@@ -0,0 +1,118 @@
+# WritableStream controller AbortSignal Explainer
+
+
+## Introduction
+
+The streams APIs provide ubiquitous, interoperable primitives for creating, composing, and consuming streams of data.
+
+This change permits an underlying sink to rapidly abort an ongoing write or close when requested by the writer.
+
+Previously, when `writer.abort()` was called, a long-running write would still have to continue to completion before
+the stream could be aborted. With this change, the write can be aborted immediately.
+
+An underlying sink which doesn't observe the `controller.signal` will continue to have the existing behavior.
+
+In addition to being exposed to streams authored in JavaScript, this facility will also be used by platform-provided
+streams such as [WebTransport](https://w3c.github.io/webtransport/).
+
+
+## API Proposed
+
+On [WritableStreamDefaultController](https://streams.spec.whatwg.org/#writablestreamdefaultcontroller)
+(the controller argument that is passed to underlying sinks):
+
+*   [abortReason](https://streams.spec.whatwg.org/#writablestreamdefaultcontroller-abortreason)
+    *   The argument passed to `writable.abort()` or `writer.abort()`. Undefined if no argument was passed or `abort()`
+    hasn't been called.
+*   [signal](https://streams.spec.whatwg.org/#writablestreamdefaultcontroller-signal)
+    *   An AbortSignal. By using `signal.addEventListener('abort', …)` an underlying sink can abort the pending write
+    or close operation when the stream is aborted.
+
+The WritableStream API does not change. Instead, the existing `abort()` operation will now signal abort.
+
+
+## Examples
+
+These are some examples of Javascript which can be used for writable streams once this is implemented:
+
+In this example, the underlying sink write waits 1 second to simulate a long-running operation. However, if abort() is
+called it stops immediately.
+
+
+```javascript
+const ws = new WritableStream({
+  write(controller) {
+    return new Promise((resolve, reject) => {
+      setTimeout(resolve, 1000);
+      controller.signal.addEventListener('abort',
+        () => reject(controller.abortReason()));
+});
```

Comment:
I think we're missing two '}' here.

Reply:
Done, I've added the correct parentheses/brackets here.